lib/xla.ex

```diff
@@ -56,6 +56,13 @@ defmodule XLA do
     target
   end
 
+  defp infer_xla_target() do
+    if nvcc = System.find_executable("nvcc") do
+      {output, 0} = System.cmd(nvcc, ["--version"])
+      if output =~ "release 12.", do: "cuda12"
+    end
```

Comment:
In case we don't want to assume it returning something successful.

```suggestion
    with nvcc when nvcc != nil <- System.find_executable("nvcc"),
         {output, 0} <- System.cmd(nvcc, ["--version"]) do
      if output =~ "release 12.", do: "cuda12"
    else
      _ -> nil
    end
```